3.analyze_data/nbconverted/log_reg_class_balanced_areashape_analyze_well_predicted_probabilities.py

```diff
@@ -119,30 +120,118 @@ def perform_ks_test(_dmso_probs, _treatment_probs):
     return zip(["comparison_metric_value", "p_value"], [stat, p_value])
 
 
-# ## Defining tests and aggregation metric names
+# ## Process the data
 
 # In[6]:
 
 
+# Fill blank broad samples in the broad_sample column with DMSO.
+# These samples are represented as DMSO in the platemap, but as nans when loaded as a DataFrame
+treatment_data["compound"]["platemap"]["broad_sample"].fillna("DMSO", inplace=True)
+treatment_data["compound"]["metadata"]["broad_sample"].fillna("DMSO", inplace=True)
+
+
+# ## Combine the model probabilty and plate data
+
+# In[7]:
+
+
+# Columns names to drop after merging data
+drop_cols = ["Assay_Plate_Barcode", "well_position"]
+
+def combine_meta(probadf):
```

Comment:
for this function consider moving to the top of this script and adding a docstring + type hints